api/src/main/java/uk/gov/hmcts/bar/api/controllers/payment/PaymentInstructionController.java

```diff
@@ -72,19 +79,57 @@ public PaymentInstructionController(PaymentInstructionService paymentInstruction
 		} else {
-			PaymentInstructionSearchCriteriaDto paymentInstructionSearchCriteriaDto = PaymentInstructionSearchCriteriaDto
-					.paymentInstructionSearchCriteriaDto().status(status)
-					.startDate(startDate == null ? null : startDate.atStartOfDay())
-					.endDate(endDate == null ? null : endDate.atTime(LocalTime.now())).payerName(payerName)
-					.chequeNumber(chequeNumber).postalOrderNumer(postalOrderNumber).dailySequenceId(dailySequenceId)
-					.allPayInstructionId(allPayInstructionId).paymentType(paymentType).action(action).build();
+			PaymentInstructionSearchCriteriaDto paymentInstructionSearchCriteriaDto =
+                createPaymentInstructionCriteria(status, startDate, endDate, payerName, chequeNumber, postalOrderNumber,
+                    dailySequenceId, allPayInstructionId, paymentType, action);
 
 			paymentInstructionList = paymentInstructionService
 					.getAllPaymentInstructions(paymentInstructionSearchCriteriaDto);
 		}
 		return Util.updateStatusAndActionDisplayValue(paymentInstructionList);
 	}
 
+    @ApiOperation(value = "Get all current payment instructions", notes = "Get all current payment instructions for a given site.",
+        produces = "application/json, text/csv")
+    @ApiResponses(value = {@ApiResponse(code = 200, message = "Return all current payment instructions for a given user"),
+        @ApiResponse(code = 404, message = "Payment instructions not found"),
+        @ApiResponse(code = 500, message = "Internal server error")})
+    @ResponseStatus(HttpStatus.OK)
+    @GetMapping("/users/{id}/payment-instructions")
+    public List<PaymentInstruction> getPaymentInstructionsByIdamId(
+        @PathVariable("id") String id,
+        @RequestParam(name = "status", required = false) String status,
+        @RequestParam(name = "startDate", required = false) @DateTimeFormat(pattern = "ddMMyyyy") LocalDate startDate,
+        @RequestParam(name = "endDate", required = false) @DateTimeFormat(pattern = "ddMMyyyy") LocalDate endDate,
+        @RequestParam(name = "payerName", required = false) String payerName,
+        @RequestParam(name = "chequeNumber", required = false) String chequeNumber,
+        @RequestParam(name = "postalOrderNumber", required = false) String postalOrderNumber,
+        @RequestParam(name = "dailySequenceId", required = false) Integer dailySequenceId,
+        @RequestParam(name = "allPayInstructionId", required = false) String allPayInstructionId,
+        @RequestParam(name = "caseReference", required = false) String caseReference,
+        @RequestParam(name = "paymentType", required = false) String paymentType,
+        @RequestParam(name = "action", required = false) String action) {
+
+        List<PaymentInstruction> paymentInstructionList = null;
+        String loggedInUserId = barUserService.getCurrentUserId();
+
+        if (!id.equals(loggedInUserId)){
+            throw new AccessDeniedException("failed to identify user");
```

Comment:
1. Do we need to add this restriction for retrieval of the payment? 
2. Or this restriction should only be in place when status is in DRAFT?
3. Should it say that, this user is not allowed to view this payment?

Reply:
Also a delivery manager should be able to see other user's payments where these are in specific status. My feeling is that  we are better off not adding this check at all.